tests/console/journalctl.pm

```diff
@@ -73,102 +73,177 @@ sub reboot {
     my ($self) = @_;
     power_action('reboot', textmode => 1);
     $self->wait_boot(bootloader_time => 300);
-    select_console 'root-console';
+    $self->select_serial_terminal;
+}
+
+sub get_current_boot_id {
+    my $boot_list = shift;
+    push @{$boot_list}, script_output('sed "s/-//g" /proc/sys/kernel/random/boot_id');
+}
+
+sub write_test_log_entries {
+    my $entries = shift;
+    foreach (keys(%{$entries})) {
+        assert_script_run("systemd-cat --priority=$_ --identifier=batman echo $entries->{$_}");
+    }
+}
+
+sub read_test_log_entries {
+    my ($entries, $boots) = @_;
+    foreach my $bootid (@{$boots}) {
+        foreach (keys(%{$entries})) {
+            assert_script_run("journalctl --boot=$bootid --identifier=batman --priority=$_ --output=short | grep $entries->{$_}");
+            assert_script_run("grep $entries->{$_} ${\ SYSLOG }") if is_sle;
+        }
+    }
 }
 
 sub run {
     my ($self) = @_;
     $self->select_serial_terminal;
-    zypper_call 'in socat';
-    # Test for bsc#1063066
-    if (script_run('command -v man') == 0) {
-        my $output = script_output('man -P cat journalctl');
-        record_soft_failure('bsc#1063066 - broken manpage') if ($output =~ m/\s+\.SH /);
+    my %log_entries = (
+        info  => q{'We need to call batman'},
+        err   => q{'We NEED to call the batman NOW'},
+        emerg => q{'CALL THE BATMAN NOW!1!! AARRGGH!!'}
+    );
+    my @boots;
+
+    # create dropin directory for further journal.conf updates
+    assert_script_run "mkdir -p ${\ DROPIN_DIR }/";
+    # Test for possible resurrection of bsc#1063066
+    die "Systemd: broken manpage!\n"
+      if ((script_run('command -v man') == 0) && (script_output('man -P cat journalctl') =~ m/\s+\.SH /));
+
+    # Enable persistent journal or check if it is enabled by default in opensuse
+    # journald.conf is almost identical for opensuse and sle
+    # default settings Storage=Auto behaves like "persistent" if the */var/log/journal* directory exists,
+    # and "volatile" otherwise (the existence of the directory controls the storage mode).
+    # Other configuration changes should be overridden using _drop-in_ file
+    # To enable persistent logging in opensuse, we use systemd-logger.rpm that creates */var/log/journal/* directory
+    get_current_boot_id \@boots;
+    if (is_opensuse) {
+        assert_script_run 'rpm -q systemd-logger';
+        assert_script_run "rpm -q --conflicts systemd-logger | tee -a /dev/$serialdev | grep syslog";
+    } else {
+        script_run("journalctl --boot=-1 | grep 'no persistent journal was found'") or die "Persistent journal should not be enabled by default on SLE!\n";
```

Comment:
just from my personal experience `or` might be tricky in perl and I always recommend to use `||` . they really not always behave the same ...

Reply:
I am using `or` due to lower precedense, but on the other hand the left side have parenthesis so `||` is fine over here as well.

https://perldoc.perl.org/perlop#List-Operators-(Rightward)